lib/hex_web/store/s3.ex

```diff
@@ -1,36 +1,24 @@
 defmodule HexWeb.Store.S3 do
   @behaviour HexWeb.Store
-  @max_attempts 10
-
-  defmacrop s3_config(opts) do
-    quote do
-      {:config,
-        unquote(opts[:url]),
-        unquote(opts[:access_key_id]),
-        unquote(opts[:secret_access_key]),
-        :virtual_hosted}
-    end
-  end
+  alias ExAws.S3
 
   def list_logs(prefix) do
     list(:logs_bucket, prefix)
   end
 
   def get_logs(key) do
-    bucket = Application.get_env(:hex_web, :logs_bucket) |> String.to_char_list
-    key = String.to_char_list(key)
-    result = retry(fn -> :mini_s3.get_object(bucket, key, [], config()) end)
-    result[:content]
+    Application.get_env(:hex_web, :logs_bucket)
+    |> S3.get_object!(key)
+    |> Map.get(:body)
```

Comment:
The get_object! Ensures that get_object returned {:ok, resp} indicating a
2XX status code.  resp in such a situation is guaranteed to have a body
key.

On Thursday, July 2, 2015, Michael Schaefermeyer <notifications@github.com>
wrote:

> In lib/hex_web/store/s3.ex
> <https://github.com/hexpm/hex_web/pull/129#discussion_r33803436>:
>
> >
> >    def list_logs(prefix) do
> >      list(:logs_bucket, prefix)
> >    end
> >
> >    def get_logs(key) do
> > -    bucket = Application.get_env(:hex_web, :logs_bucket) |> String.to_char_list
> > -    key = String.to_char_list(key)
> > -    result = retry(fn -> :mini_s3.get_object(bucket, key, [], config()) end)
> > -    result[:content]
> > +    Application.get_env(:hex_web, :logs_bucket)
> > +    |> S3.get_object!(key)
> > +    |> Map.get(:body)
>
> Wondering wether this should be Dict.fetch!(:body) in the spirit of
> raising when missing...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hexpm/hex_web/pull/129/files#r33803436>.
>